lib/declare_schema/dsl.rb

```diff
@@ -32,8 +32,11 @@ def field(name, type, *args, **options)
       @model.declare_field(name, type, *(args + [@options.merge(options)]))
     end
 
-    def method_missing(type, name, *args)
-      field(name, type, *args)
+    # TODO: make [:required] just another option. Either 'required: true] or 'optional: false'?
+    def method_missing(*args, **options)
+      args.count(&:itself) >= 2 or raise ::ArgumentError, "fields in declare_schema block must be declared as: type name, [:required], options (got #{args.inspect}, #{options.inspect})"
```

Comment:
NIT: Missing comma in error message
```suggestion
      args.count(&:itself) >= 2 or raise ::ArgumentError, "fields in declare_schema block must be declared as: type, name, [:required], options (got #{args.inspect}, #{options.inspect})"
```

Reply:
Actually there is no comma there because the `type` is the method name. As in
```
integer :percent_allowed
```